java/code/src/com/redhat/rhn/manager/system/SystemManager.java

```diff
@@ -2227,7 +2228,7 @@ public byte[] createProxyContainerConfig(User user, String proxyName, Integer pr
             httpdConfig.put("server_key", proxyPair.getKey());
         }
         catch (IllegalArgumentException err) {
-            throw new RhnRuntimeException("Certificate check failure: " + err.getMessage());
+            throw new SSLCheckException("Certificate check failure");
```

Comment:
I would probably not be that radical in the change: there probably are cases where the error message brings value. But Since you create a new exception maybe this could be used to provide more user-friendly messages and hide the certificates data?